src/nvim/ops.c

```diff
@@ -1053,7 +1053,7 @@ static void pbyte(pos_T lp, int c)
 {
   assert(c <= UCHAR_MAX);
   char *p = ml_get_buf_mut(curbuf, lp.lnum);
-  colnr_T len = curbuf->b_ml.ml_line_len;
+  colnr_T len = ml_get_buf_len(curbuf, lp.lnum);
```

Comment:
Was this suppose to consider `NUL`?

Reply:
It's a bit inconsistent. It seems that `len - 2` considers `NUL`, but `lp.col >= len` doesn't...